src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,6 +33,13 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        String copyTrimmedAddress = new String(trimmedAddress);
+        String[] addressParts = copyTrimmedAddress.split(",");
+        this.block = new Block(addressParts[0].trim().substring(2));
+        this.street = new Street(addressParts[1].trim());
+        this.unit = new Unit(addressParts[2].trim());
+        this.postalCode = new PostalCode(addressParts[3].trim());
```

Comment:
Avoid the use of 'magic' numbers. Declare variables for the position of each part.